runelite-client/src/main/java/net/runelite/client/plugins/mining/MiningPlugin.java

```diff
@@ -174,4 +239,77 @@ public void onWallObjectSpawned(WallObjectSpawned event)
 			}
 		}
 	}
+
+	@Subscribe
+	public void onAnimationChanged(AnimationChanged event)
+	{
+		Player local = client.getLocalPlayer();
+
+		if (event.getActor() != local)
+		{
+			return;
+		}
+
+		int playerAnimationId = local.getAnimation();
+
+		switch (playerAnimationId)
+		{
+			case MINING_BRONZE_PICKAXE:
+			case MINING_IRON_PICKAXE:
+			case MINING_STEEL_PICKAXE:
+			case MINING_BLACK_PICKAXE:
+			case MINING_MITHRIL_PICKAXE:
+			case MINING_ADAMANT_PICKAXE:
+			case MINING_RUNE_PICKAXE:
+			case MINING_DRAGON_PICKAXE:
+			case MINING_DRAGON_PICKAXE_UPGRADED:
+			case MINING_DRAGON_PICKAXE_OR:
+			case MINING_INFERNAL_PICKAXE:
+			case MINING_3A_PICKAXE:
+			case MINING_CRYSTAL_PICKAXE:
+			case DENSE_ESSENCE_CHIPPING:
+				mining = true;
```

Comment:
can't you simply make method that checks what is current player's animation instead of adding boolean and checking it in anim id? Would be simpler imo (e.g simple list.contains(playerAnimId) and make method isMining that does that)

Reply:
Changing it to boolean is so insignificat performance boost that it doesnt matter, and just doing contains on list is way cleaner and simpler.